### ISSUE TITLE
Allow for alternative Prefix Trimming order

### ISSUE THREAD

**Issue by spencerr** [Azure/AppConfiguration-DotnetProvider#573]
The current implementation of prefix trimming is backed by a SortedSet, removing the ability to trim prefixes in a specific order specified by the user. 

I currently use a single App Configuration instance to serve configurations for multiple applications. The UI experience is not the best for displaying groups configurations by Label, so I utilize a prefix of `{app-name}/` or `shared/` and such to group these configurations and display in the hierarchy view. The downfall of this is when I need to trim the prefixes for when an app specifically overrides one of the shared values, I am unable to because of the automatic ordering of the SortedSet.

**Comment by spencerr** [comment]
Thanks for the reply @zhiyuanliang-ms 

Yes, your scenario matches what I am describing.

In our setup, we have various "groups" of variables that we may share between applications. Think items like redis connection information, application urls, etc. We'd like centralize configurations in a single place so we are not duplicating these values across different app configuration instances per application, and to opt-in or control the shared variables an application is given.

Another additional limitation we run into is snapshots having a limit of 3 filters that can be applied. Right now I filter for the `{app-name}/*`, and `shared/*` and accept that all shared configurations will be retrieved, were as we'd like to be more granular with the items being pulled in and to be able to use the snapshots.

As far as UI experience, updating the hierarchy view to be able to group by Label and/or separator would be beneficial.

**Comment by zhiyuanliang-ms** [comment]
Hi, @spencerr 

Here is the screenshot of my app config store
<img width="344" alt="image" src="https://github.com/user-attachments/assets/6694c888-3472-4b9b-8aa9-86e3654033fd">

By adjusting the order of the `Select` call, you can specify the order how the keyvalue with different prefix will be loaded. Will this solve your problem?
<img width="361" alt="image" src="https://github.com/user-attachments/assets/a99c5de5-0645-4c8b-a6f6-53fe1942f6ae">

**Comment by zhiyuanliang-ms** [comment]
BTW, my previous comment is wrong.

> ~~IMO, label should be used for the scenario that you have different sets of configuration for different applications. Label should be used to group configurations.~~

According to the [best practice](https://learn.microsoft.com/en-us/azure/azure-app-configuration/howto-best-practices#key-groupings), using different prefixs (as different namespaces) to group key-values is correct. (My previous comment is wrong. Sorry for the mistake.)

@spencerr

**Comment by amerjusupovic** [comment]
> As far as UI experience, updating the hierarchy view to be able to group by Label and/or separator would be beneficial.

Does the option to filter by label on the configuration explorer page work for your scenario or did you mean something different?

<img width="464" alt="image" src="https://github.com/user-attachments/assets/ab19abd9-6f92-40b1-84cb-8b3bdb6db557">

**Comment by spencerr** [comment]
@zhiyuanliang-ms 
The ordered select does seem to work, however it does not work with snapshots unfortunately.

@amerjusupovic Filter by the label could work, grouping things does make things a bit easier to visualize. Best practices does seem to mention prefixes being used for separating variables per application however.

**Comment by zhiyuanliang-ms** [comment]
Hi, @spencerr 
> The ordered select does seem to work, however it does not work with snapshots unfortunately.

Could you try to create two snapshots for each prefix like this:
<img width="350" alt="image" src="https://github.com/user-attachments/assets/0cb78994-1eca-472f-9382-1b9fa1b09717">


Instead of having one snapshot which used two key(prefix) filters like this:
<img width="321" alt="image" src="https://github.com/user-attachments/assets/be3f1eb1-f19f-4887-876f-9799a766ab26">

**Comment by samsadsam** [comment]
Hey @spencerr hope your day is going well. I will recap the issue here and will take as an example this store:

![Image](https://github.com/user-attachments/assets/99c45bc8-e363-4a14-8be7-5d9382cc0412)

This store has 2 snapshots. 
- `app1` snapshot contains `app1/key1` kv item
- `shared` snapshot contains `shared/key1` kv item

![Image](https://github.com/user-attachments/assets/c4f9fb31-df44-44d5-a7d3-163d4900f5b5)


You mentioned that you were unable to override some shared settings with a specific app settings. For a scenario like that, the recommendation is to have your `Select`/`SelectSnapshot` statements in this order:

```
builder.AddAzureAppConfiguration(options =>
{
    options.Connect(...)
           .Select("shared/*")
           .Select("app1/*")
           .TrimKeyPrefix("shared/")
           .TrimKeyPrefix("app1/");
});
```

Finally, if you try the snippet of code below, the output would be `val=app1/key1` as you wanted. The `app1` setting `key1` overrides the `shared` setting `key 1`.

```
static void Main(string[] args)
{
    var builder = new ConfigurationBuilder();

    builder.AddAzureAppConfiguration(options =>
    {
        options.Connect(Environment.GetEnvironmentVariable("AAC_CONNECTION_STRING"))
               .SelectSnapshot("shared")
               .SelectSnapshot("app1")
               .TrimKeyPrefix("shared/")
               .TrimKeyPrefix("app1/");
    });

    var config = builder.Build();

    Console.WriteLine(config["key1"]);
}
```

In summary, it is the order of `Select` that matters in your scenario.

I will close this issue. Please re-open it if you have any extra questions/comments/concerns :smiley:
